mathinspector/__init__.py

```diff
@@ -157,7 +158,58 @@ def __init__(self, *args, debug=False, disable=[]):
 			self.modules.addfile(i)
 
 
-def main(*args, **kwargs):
+def main(argv=None):
+	if argv is None:
+		argv = sys.argv[1:]
+	args = []
+	kwargs = {}
+	i = 0
+
+	while i < len(argv):
+		arg = argv[i]
+		if arg[:2] != "--":
+			args.append(argv[i])
+		elif "=" in arg:
+			key, value = arg[2:].split("=")
+			kwargs[key] = value
+		elif i + 1 < len(argv) and argv[i+1] and argv[i+1][:2] != "--":
+			kwargs[arg[2:]] = argv[i+1]
+			i += 1
+		else:
+			kwargs[arg[2:]] = True
+		i += 1
+
+	if "help" in kwargs and kwargs["help"] is True:
+		print ("""usage: mathinspector [files] ... [options] ...
+
+Options and arguments:
+[files]...       : a list of files with either a .math or .py extension.  The
+                  .math file will be loaded, and all of the .py files will be
+                  added to the current project
+
+--help obj       : view the documentation for obj as if you called help(obj)
+                  in mathinspector
+
+--new            : starts a new project and resets the state of the app.
+                  This flag will overwrites the autosave file with a blank
+                  file
+
+--disable[=opts] : mathinspector overrides a lot of builtins (e.g. help(),
+                  print()) which may cause undesirable behaviour, this flag
+                  disables these features
+                    opts are 'print', 'traceback', 'stderr'
+                    e.g. mathinspector --disable=print,stderr
+
+--debug          : prints log messages to the command line used to launch the
+                  app.  Useful for debugging issues when something isn't
+                  working properly, or while working on the mathinspector
+                  source code
+""")
+	else:
+		run(*args, **kwargs)
+
+
+def run(*args, **kwargs):
```

Comment:
What is the purpose of having another function in this file (in addition to `main`) which does nothing other than allow the `main` function to parse the argv?  Are there not 1 too many functions here?  Wouldn't a good solution be to not have a file `__main__` which does nothing other than call a function in this file?